### ISSUE TITLE
GRPO训练出现warning信息：Rollout logprobs count does not match completion tokens count

### ISSUE THREAD

**Issue by yingzhao27** [modelscope/ms-swift#6890]
**Describe the bug**
[WARNING:swift] Rollout logprobs count (208) does not match completion tokens count (207). Skipping rollout importance sampling for this batch.
[WARNING:swift] Rollout logprobs count (255) does not match completion tokens count (254). Skipping rollout importance sampling for this batch.
......
[WARNING:swift] Rollout logprobs count (155) does not match completion tokens count (154). Skipping rollout importance sampling for this batch.

华为NPU上，启动GRPO训练（gsm8k数据）日志中每个step均出现上述warning，启动方式：
USE_OPTIMIZED_MODEL=0  ASCEND_RT_VISIBLE_DEVICES=4 swift rollout --model /yunpan/zhaoyin/model/Qwen2___5-0___5B-Instruct --host 0.0.0.0 --port 8999 --template_backend swift --model_type qwen2_5 --vllm_tensor_parallel_size 1

ASCEND_RT_VISIBLE_DEVICES=5,6 NPROC_PER_NODE=2 swift rlhf --rlhf_type grpo --model /yunpan/zhaoyin/model/Qwen2___5-0___5B-Instruct --reward_funcs string_sim --use_vllm true --vllm_mode server --vllm_server_host 127.0.0.1 --vllm_server_port 8999 --train_type full --torch_dtype bfloat16 --dataset ./gsm8k_data/train.jsonl --split_dataset_ratio 0 --save_steps 200 --max_completion_length 4096 --max_length 512 --num_train_epochs 1 --per_device_train_batch_size 1 --learning_rate 2e-5 --gradient_accumulation_steps 1 --logging_steps 1 --warmup_ratio 0.05 --dataloader_num_workers 4 --dataset_num_proc 4 --num_generations 2 --temperature 0.3 --deepspeed zero2_offload --log_completions true --num_iterations 1 --beta 0.001

**Your hardware and system info**
Ascend 910B3， torch/torch_npu=2.7.1， vllm/vllm_ascend=0.11.0

**Comment by hjh0119** [comment]
run normally on my side

can you retry with latest code?

**Comment by yingzhao27** [comment]
用最新版本的代码运行上述训练命令（template_backend=swift），确实正常了。请问这部分验证是为了解决train-infer-mismatch问题吗?      但是如果设置template_backend=‘jinja’，还是会有这个warning，这个正常吗，会不会影响训练效果。    

另外，在新代码中指定template_backend=false会报错，我是在grpo_trainer.py中_prepare_batch_inputs函数中data['messages'] = replace_assistant_response_with_ids(data['messages'],data['response_token_ids'], loss_mask)前面添加了个if template.template_backend!='jinja': ，这样不会报错。请问这个修改是否合理呢

**Comment by hjh0119** [comment]
训练的 template_backend 不推荐使用 jinja

**Comment by maoulee** [comment]
> 训练的 template_backend 不推荐使用 jinja

tokenizer's values. Updated tokens: {'bos_token_id': None}.
[2025-12-08 11:22:24,989] [WARNING] [engine.py:1406:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****
Parameter Offload - Persistent parameters statistics: param_count = 145, numel = 196096
Train:   0%|                                           | 0/1358 [00:00<?, ?it/s]INFO 12-08 11:22:28 [block_pool.py:428] Successfully reset prefix cache
INFO 12-08 11:22:28 [block_pool.py:428] Successfully reset prefix cache
/root/miniconda3/lib/python3.10/site-packages/torch/utils/checkpoint.py:85: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
/root/miniconda3/lib/python3.10/site-packages/torch/utils/checkpoint.py:85: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
[WARNING:swift] Rollout logprobs count (505) does not match completion tokens count (511). Skipping rollout importance sampling for this batch.
[WARNING:swift] Rollout logprobs count (620) does not match completion tokens count (626). Skipping rollout importance sampling for this batch.
[WARNING:swift] Rollout logprobs count (1278) does not match completion tokens count (1284). Skipping rollout importance sampling for this batch.
[WARNING:swift] Rollout logprobs count (1045) does not match completion tokens count (1051). Skipping rollout importance sampling for this batch.

我使用grpo的多轮训练，同样的脚本，3.11.0dev出错了，但是3.10.3却能正常通过

**Comment by hjh0119** [comment]
warning 不影响训练

**Comment by Vanza11** [comment]
我也出现了这个问题，请问你最后解决了吗，还是忽略这个警告？这个问题会导致reward一直上升不了吗
